### PR TITLE
Add documentation site for blugnu/test framework

### PR DESCRIPTION
**Pull request by deltics**
## Summary by CodeRabbit

* **New Features**
  * Live documentation site with homepage, deploy workflow, responsive navbar, cover hero, theme switcher (light/dark/auto), code highlighting with copy-to-clipboard, and search.

* **Documentation**
  * Comprehensive docs: Getting Started, Concepts, Usage, Migration, Advanced guides, examples (expectations, matchers, runners, panic testing, helpers, mocking, flaky/parallel/table-driven tests).

* **Chores**
  * Site build/tooling and config updates, style additions, .gitignore and markdown-lint tweaks.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+permissions:
+  contents: write
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```
[the remark this answers is not in the document]
Comment (deltics) [reply .github/workflows/docs.yml:25]: I accept the risk in using semantic versions for the clarity this provides as these dependencies are source from reputable and well maintained repositories

File: docs/content/docs/usage/table-driven-tests.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+```go
+Skip("edge case", divideCase{...})
+```
+
+Or set a `Skip bool` / `skip bool` field on the struct.
+
+{{< alert title="Warning" color="warning" >}}
+`Skip` cases always cause the test suite to fail, even when the case itself passes.
+{{< /alert >}}
```
[the remark this answers is not in the document]
Comment (deltics) [reply docs/content/docs/usage/table-driven-tests.md:168]: The warning is correct.  Although t.Skip does not cause a test to fail, the Testcases runner includes a step which emits a warning if any cases are marked to be skipped.  Any warning causes the test to fail.  this is forced to ensure that the warning is actually emitted since t.Logf output is only output if there are other failures in a test.

The API documentation will be clarified separately (this commit is strictly for docs site creation/changes).

File: docs/layouts/partials/navbar.html [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      {{ if (gt (len .Site.Home.Translations) 0) -}}
+      <li class="nav-item td-navbar__lang-menu">
+        {{ partial "navbar-lang-selector.html" . -}}
+      </li>
+      {{ end -}}
+      {{- $darkMode := partialCached "dark-mode-config.html" "dark-mode-global" -}}
+      {{ if $darkMode.showMenu -}}
+      <li class="nav-item td-navbar__light-dark-menu">
+        {{ partial "theme-toggler" . }}
+      </li>
```
[the remark this answers is not in the document]
Comment (deltics) [reply docs/layouts/partials/navbar.html:67]: the docsy theme provides this partial
